plugins/outputs/azure_monitor_logs/azure_monitor_logs.go

```diff
@@ -0,0 +1,253 @@
+)
+
+// AzLogAnalytics contains information about a azure log analytics service metadata
+type AzLogAnalytics struct {
+	CustomerID string `toml:"customer_id"`
+	SharedKey  string `toml:"shared_key"`
+
+	NamespacePrefix string            `toml:"namespace_prefix"`
+	ClientTimeout   internal.Duration `toml:"timeout"`
+
+	serviceURL string
+	client     *http.Client
+}
+
+// Initializes the plugin
+func (a *AzLogAnalytics) Init() error {
+	if a.CustomerID == "" {
+		return fmt.Errorf("customer_id not configured")
+	}
+
+	if a.SharedKey == "" {
+		return fmt.Errorf("shared_key not configured")
+	}
+
+	if len(a.NamespacePrefix) > 25 {
+		return fmt.Errorf("namespace_prefix length is greater than 25 characters")
+	}
+
+	if a.ClientTimeout.Duration == 0 {
+		a.ClientTimeout.Duration = defaultClientTimeout
+	}
+
+	match, err := regexp.MatchString(namespacePrefixRegex, a.NamespacePrefix)
+	if err != nil {
+		return err
+	} else if !match {
+		return fmt.Errorf("namespace_prefix contains invalid characters")
+	}
+
+	return nil
+}
+
+// Connect validates connectivity
+func (a *AzLogAnalytics) Connect() error {
+
+	err := a.Init()
+	if err != nil {
+		return err
+	}
+
+	client, err := a.createClient(context.Background())
+	if err != nil {
+		return err
+	}
+
+	a.client = client
+	a.serviceURL = fmt.Sprintf(baseURL, a.CustomerID)
+
+	return nil
+}
+
+// Close shuts down an any active connections
+func (a *AzLogAnalytics) Close() error {
+	a.client.CloseIdleConnections()
+
+	return nil
+}
+
+// Description provides a description of the plugin
+func (a *AzLogAnalytics) Description() string {
+	return "A plugin that can transmit metrics to Azure Log Analytics"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (a *AzLogAnalytics) SampleConfig() string {
+	return sampleConfig
+}
+
+// Write writes metrics to the remote endpoint
+func (a *AzLogAnalytics) Write(metrics []telegraf.Metric) error {
+
+	objects := make(map[string][]interface{}, len(metrics))
+	for _, metric := range metrics {
+		m, name := a.createObject(metric)
+		objects[name] = append(objects[name], m)
+	}
+
+	for name, data := range objects {
+		serialized, err := json.Marshal(data)
+		if err != nil {
+			return err
+		}
+
+		if err := a.write(name, serialized); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (a *AzLogAnalytics) write(logType string, reqBody []byte) error {
+
+	dateString := time.Now().UTC().Format(time.RFC1123)
+	dateString = strings.Replace(dateString, "UTC", "GMT", -1)
+
+	stringToHash := httpMethod + "\n" + 
+					strconv.Itoa(utf8.RuneCount(reqBody)) + "\n" +
+					contentType + "\n" + "x-ms-date:" +
+					dateString + "\n/api/logs"
+
+	signature, err := a.buildSignature(stringToHash)
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest(httpMethod, a.serviceURL, bytes.NewReader(reqBody))
+	if err != nil {
+		return err
+	}
+
+	tableName := a.NamespacePrefix + underscoreToCaml(logType)
```

Comment:
Can you please do this in `Write()` so we only deal with actually managing the sending in here!?